From f38ab53626ffd56c99bcce3138eba5b9b0c5f69a Mon Sep 17 00:00:00 2001
From: Michal Rostecki <vadorovsky@protonmail.com>
Date: Tue, 26 Dec 2023 23:46:06 +0100
Subject: [PATCH] pci: apple: Initialize and increment the port counter

clang rightfully issued a `-Wuninitialized` warning, because `i`
was never initialized. It was also never incremented, so the error
message was incorrect.

Fix that by initializing and incrementing `i` on each iteration.

Signed-off-by: Michal Rostecki <vadorovsky@protonmail.com>
---
 drivers/pci/pcie_apple.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/pcie_apple.c b/drivers/pci/pcie_apple.c
index 21bafba3b0..c41abdad3d 100644
--- a/drivers/pci/pcie_apple.c
+++ b/drivers/pci/pcie_apple.c
@@ -336,7 +336,7 @@ static int apple_pcie_probe(struct udevice *dev)
 	struct apple_pcie_priv *pcie = dev_get_priv(dev);
 	fdt_addr_t addr;
 	ofnode of_port;
-	int i, ret;
+	int i = 0, ret;
 
 	pcie->hw = (struct reg_info *)dev_get_driver_data(dev);
 
@@ -363,6 +363,7 @@ static int apple_pcie_probe(struct udevice *dev)
 			dev_err(pcie->dev, "Port %d setup fail: %d\n", i, ret);
 			return ret;
 		}
+		i++;
 	}
 
 	return 0;
-- 
2.39.3 (Apple Git-145)

